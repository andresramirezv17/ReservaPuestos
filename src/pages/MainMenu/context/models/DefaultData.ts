export interface DefaultData {
  building: {
    name: string;
    idBuilding: number;
  };
  floor: {
    name: string;
    idBuilding: number;
  };
  section: {
    name: string;
    idBuilding: number;
  };
}
