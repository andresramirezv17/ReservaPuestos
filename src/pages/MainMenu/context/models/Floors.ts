export interface FloorFound {
  idFloor: number;
  idBuilding: number;
  name: string;
}

export interface Floors {
  id?: string;
  floorsResult?: FloorFound[];
}
