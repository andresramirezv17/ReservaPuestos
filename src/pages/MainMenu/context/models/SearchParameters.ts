export interface SearchParameters {
  nameBuilding: string;
  floor: string;
  section: string;
  isAvailable: boolean;
  date: string;
  initialHour: string;
  endHour: string;
}
