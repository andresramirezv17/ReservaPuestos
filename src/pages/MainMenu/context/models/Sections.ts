export interface SectionFound {
  idSection: number;
  idBuilding: number;
  name: string;
}

export interface Sections {
  id?: string;
  sectionsResult?: SectionFound[];
}
