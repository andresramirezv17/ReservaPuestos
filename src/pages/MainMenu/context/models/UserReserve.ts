export interface UserReserve {
  idReserve?: number;
  building: string;
  floor: string;
  section: string;
  date: string;
  initialHour: string;
  workplace: string;
  endHour: string;
}
