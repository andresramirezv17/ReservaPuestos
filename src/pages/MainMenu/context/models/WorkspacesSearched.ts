export interface TotalWorkspaces {
  id?: string;
  idBuilding?: number;
  idFloor?: number;
  idSection?: number;
  available?: boolean;
  initialHour: string;
  endHour: string;
}

export interface WorkspacesSearched {
  id?: number;
  workplaces?: TotalWorkspaces[];
}
