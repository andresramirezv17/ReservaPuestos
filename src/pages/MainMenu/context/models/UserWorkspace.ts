import { UserReserve } from './UserReserve';

export interface UserWorkspace {
  id: number;
  building: string;
  floor: string;
  section: string;
  date: string;
  initialHour: string;
  workplace: string;
  endHour: string;
  isChekin: boolean;
  idWorkSpace?: number;
}
