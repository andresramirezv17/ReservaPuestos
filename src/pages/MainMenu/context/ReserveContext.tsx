import React, { createContext, useState } from 'react';
import moment from 'moment';
import { WorkspacesSearched } from './models/WorkspacesSearched';
import { Floors } from './models/Floors';
import { Sections } from './models/Sections';
import * as ReserveServices from '../services/WorkspaceServices';

export interface WorkSpaceParameter {
  workspacesSearched?: WorkspacesSearched[];
  floors?: Floors;
  sections?: Sections;
}

export interface propsSection {
  idFloor: number;
  idBuilding: number;
  name: string;
}

export const useStateContainer = (initialState: WorkSpaceParameter) => {
  const [workplaces, setworkplaces] = useState(
    initialState.workspacesSearched || [],
  );

  const [floors, setfloors] = useState(
    initialState.floors || { floorsResult: [] },
  );
  const [sections, setsections] = useState(
    initialState.sections || { sectionsResult: [] },
  );
  const [day, setday] = useState('');
  const [building] = useState('Torre Central 2');
  const [floor, setfloor] = useState('Piso 1');
  const [section, setsection] = useState('SecciÃ³n A');
  const [fecha, setfecha] = useState(
    moment(new Date()).add(1, 'days').format('DD/MM/YYYY'),
  );
  const [initHour, setinitHour] = useState('7:00');
  const [endHour, setendHour] = useState('19:00');

  const searchWorkplaces = () => {
    ReserveServices.getSearchedWorkplaces(floor, section).then(
      (resultSearch) => {
        setworkplaces(resultSearch);
      },
    );
  };
  const obtainSections = () => {
    ReserveServices.getSections().then((resultSections) =>
      setsections(resultSections),
    );
  };

  const obtainFloors = () => {
    ReserveServices.getFloors().then((resultFloor) => {
      setfloors(resultFloor);
      obtainSections();
    });
  };

  const changeParameter = (parameter: string, value: string) => {
    if (parameter === 'piso') {
      setfloor(value);
    }
    if (parameter === 'day') {
      setday(value);
    }
    if (parameter === 'section') {
      setsection(value);
    }
    if (parameter === 'date') {
      setfecha(value);
    }
    if (parameter === 'inithour') {
      setinitHour(value);
    }
    if (parameter === 'endhour') {
      setendHour(value);
    }
  };

  return {
    data: {
      workplaces,
      floors,
      sections,
      building,
      floor,
      section,
      fecha,
      initHour,
      endHour,
      day,
    },
    mutations: {
      searchWorkplaces,
      obtainFloors,
      obtainSections,
      changeParameter,
    },
  };
};

export const ReserveContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

export interface SearchWorkspaceProvider {
  initialState?: WorkSpaceParameter;
}

export const ReserveProvider: React.FC<SearchWorkspaceProvider> = ({
  children,
  initialState = {},
}) => {
  const contextValue = useStateContainer(initialState);
  return (
    <ReserveContext.Provider value={contextValue}>
      {children}
    </ReserveContext.Provider>
  );
};
