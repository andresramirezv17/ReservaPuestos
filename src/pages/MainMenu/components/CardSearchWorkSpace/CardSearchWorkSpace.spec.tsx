import React from 'react';
import { CardSearchWorkSpace, cardSearchProps } from '.';
import { render, fireEvent } from '@testing-library/react';
import { ReserveContext } from '../../context/ReserveContext';
import { WorkspacesSearched } from 'pages/MainMenu/context/models/WorkspacesSearched';
import { Floors } from 'pages/MainMenu/context/models/Floors';
import { Sections } from 'pages/MainMenu/context/models/Sections';

describe('CardSearchWorkSpace test', () => {
  let props: cardSearchProps;
  let searchWorkplaces: jest.Mock;
  let obtainFloors: jest.Mock;
  let obtainSections: jest.Mock;
  let changeParameter: jest.Mock;
  let workplaces: WorkspacesSearched[],
    building: string,
    floor: string,
    section: string,
    fecha: string,
    initHour: string,
    endHour: string,
    day: string;
  let floors: Floors, sections: Sections;

  beforeEach(() => {
    searchWorkplaces = jest.fn();
    obtainFloors = jest.fn();
    obtainSections = jest.fn();
    changeParameter = jest.fn();
    props = {
      showModal: jest.fn(),
    };
    workplaces = [
      {
        id: 1,
        workplace: 'P2D1',
        available: true,
        building: 'Torre Central',
        floor: 'Piso 1',
        section: 'SecciÃ³n A',
        initialHour: '7:00',
        endHour: '5:00',
      },
    ];
  });

  it('it should match snapshot', () => {
    const { container } = render(
      <ReserveContext.Provider
        value={{
          mutations: {
            searchWorkplaces,
            obtainFloors,
            obtainSections,
            changeParameter,
          },
          data: {
            workplaces,
            floors,
            sections,
            building,
            floor,
            section,
            fecha,
            initHour,
            endHour,
            day,
          },
        }}
      >
        <CardSearchWorkSpace {...props} />
      </ReserveContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should call showmodal when button is clicked', () => {
    const { getByTestId } = render(
      <ReserveContext.Provider
        value={{
          mutations: {
            searchWorkplaces,
            obtainFloors,
            obtainSections,
            changeParameter,
          },
          data: {
            workplaces,
            floors,
            sections,
            building,
            floor,
            section,
            fecha,
            initHour,
            endHour,
            day,
          },
        }}
      >
        <CardSearchWorkSpace {...props} />
      </ReserveContext.Provider>,
    );
    const button = getByTestId('reserveBtn');
    fireEvent.click(button);
    expect(props.showModal).toHaveBeenCalled();
  });
});
