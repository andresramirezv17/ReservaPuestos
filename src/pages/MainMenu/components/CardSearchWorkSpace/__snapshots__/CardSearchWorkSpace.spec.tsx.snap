// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSearchWorkSpace test it should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Busqueda: 
    </p>
    <div
      class="makeStyles-divSearch-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      />
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
         a 
      </p>
    </div>
    <div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-mainBox-5 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div
            style="display: flex; flex-direction: row; justify-content: space-between;"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
              style="font-weight: bold;"
            >
              Puesto:
               
              <span
                style="font-weight: normal;"
              >
                P2D1
              </span>
            </p>
            <div>
              <div
                style="display: flex; flex-direction: row;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                  <path
                    d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  7:00
                   a 
                  5:00
                </p>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttonReserve-6"
                data-testid="reserveBtn"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Reservar
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
</div>
`;
